### PR TITLE
UI update

### PR DESCRIPTION
**Pull request by simonweniger**
## Summary

This PR adds SaasUI and updates the sidebar as well as the layout. It is the groundwork to allow custom user themes and more improvements on the overall UI.

I kept the core UI and themes! Made just minor style improvements. The tags in the screenshot are not in this PR, its a UI example for another PR.

Fixes

Depends on

## Test plan

-

## Screenshots

**Before**
Bildschirm­foto 2023-07-17 um 06 46 29

**After**
Bildschirm­foto 2023-07-17 um 06 50 28

## Checklist

- [x] My code follows the code style of this project and passes `make format`.
- [ ] My change requires a change to the documentation.
- [ ] I have [updated the documentation](https://docs.superagent.sh/) accordingly.
- [ ] My change has adequate unit test coverage.

### CODE REVIEW

File: ui/app/layout.js [hunk 1]
```diff
@@ -23,11 +25,9 @@ export default async function RootLayout({ children }) {
     <html lang="en">
       <body className={inter.className}>
         <Providers>
-          <Container>
-            {session && <Navigation />}
```
Comment (homanp) [inline ui/app/layout.js:27]: @simonweniger this was there to only show the Sidebar of user logged in. Does your changes take that into account?

**Review by homanp — CHANGES_REQUESTED** [review]

File: ui/app/_components/appbody.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+export default function AppBody({ children, session }) {
+  return (
+    <AppShell
+      variant="fixed"
+      minH="100vh"
+      maxH="100vh"
+      overflow="hidden"
+      sidebar={
```
Comment (simonweniger) [inline ui/app/_components/appbody.js:56]: @homanp Right I need to refactor this a bit as the AppShell needs to wrap around the whole child. Then I forgot to reimplement it in the new AppBody. So here I added it again. This now only shows the sidebar if logged in.

File: ui/app/layout.js (hunk 1 above)
Comment (simonweniger) [reply ui/app/layout.js:27]: right, i forgot to reimplement this in the refectoring... Added a commit to reimplement it. And tested login & logout flow

**Review by homanp — APPROVED** [review]

**Review by homanp — CHANGES_REQUESTED** [review]
@simonweniger did some testing and seems like the dark mode part is broken. Could you force dark mode? 

image

**Comment by simonweniger** [comment]
> @simonweniger did some testing and seems like the dark mode part is broken. Could you force dark mode?
> 
> image

I added force dark mode via a config in the theme

`const config = {
  initialColorMode: 'dark',
  useSystemColorMode: false,
}`

**Review by homanp — APPROVED** [review]

**Review by homanp — APPROVED** [review]

**Review by homanp — CHANGES_REQUESTED** [review]
image

Mobile sidebar kinda broken!

**Review by homanp — APPROVED** [review]

**Pull request merged**